docker/utils/install_dependencies.sh

```diff
@@ -117,6 +117,33 @@ mysql80)
         percona-xtrabackup-80
     )
     ;;
+mysql84)
+    if [ -z "$VERSION" ]; then
+        VERSION=8.4.0
```

Comment:
@L3o-pold @frouioui We should use the latest 8.4 release here and not 8.4.0 I think? So 8.4.3?

Reply:
8.4.3 is not compatible with bullseye :/

Reply:
@L3o-pold I think that's a very strong signal we should first upgrade to bookworm then and only add these after that step. 

It's moving, but doing it then in that order will make things a lot more easier I think.

Reply:
I can close this and contribute to you other PR if needed

Reply:
@L3o-pold Let's not do too many things in one PR, we can rebase this one on top of that once it lands? But you can branch off that PR to test more if you want to.

Reply:
@L3o-pold https://github.com/vitessio/vitess/pull/17552 is now merged which also bumped things to bookworm. So we should be good to update to that here too and use that for 8.4 as well.